imagedephi/main.py

```diff
@@ -24,12 +27,31 @@ def imagedephi() -> None:
     "input-dir", type=click.Path(exists=True, file_okay=False, readable=True, path_type=Path)
 )
 @click.argument(
-    "output-dir", type=click.Path(exists=True, file_okay=False, writable=True, path_type=Path)
+    "output-dir", type=click.Path(exists=True, file_okay=False, readable=True, path_type=Path)
 )
-def run(input_dir: Path, output_dir: Path) -> None:
-    """Run in CLI-only mode."""
-    redact_images(input_dir, output_dir)
-    click.echo("Done!")
+@click.argument("override-rules", type=click.File("r"), required=False)
+def run(input_dir: Path, output_dir: Path, override_rules: Optional[TextIO]):
+    """Redact images in a folder according to given rule sets."""
+    base_rules_bytes = pkgutil.get_data("imagedephi", "base_rules.yaml")
+    if base_rules_bytes is None:
+        raise click.ClickException("There was an error with finding the base rules.")
+    base_rule_set = build_ruleset(yaml.safe_load(base_rules_bytes))
+    override_rule_set = build_ruleset(yaml.safe_load(override_rules)) if override_rules else None
+    redact_images(input_dir, output_dir, base_rule_set, override_rule_set)
+
+
+@imagedephi.command
+@click.argument("image", type=click.Path())
+@click.argument("override-rules", type=click.File("r"), required=False)
+def redaction_plan(image: click.Path, override_rules: Optional[TextIO]) -> None:
+    """Print the redaction plan for a given image and rules."""
+    click.echo(override_rules)
```

Comment:
Rather than implement all this rule-loading logic in every sub-command, we can add it to the root command, then pass the parsed data as Click "context" to every subcommand. See https://click.palletsprojects.com/en/8.1.x/complex/